src/Microsoft.PackageManagement/Implementation/PackageManagementService.cs

```diff
@@ -1037,11 +1037,16 @@ private IMetaProvider GetMetaProviderObject(IHostApi request)
                 throw new ArgumentNullException("request");
             }
 
+            var providerAssemblies = Enumerable.Empty<string>();
+
+#if !CORECLR
             //looks like registry needs to be here for supporting .msi packages
-            var providerAssemblies = (_initialized ? Enumerable.Empty<string>() : _defaultProviders)
+             providerAssemblies = (_initialized ? Enumerable.Empty<string>() : _defaultProviders)
                 .Concat(GetProvidersFromRegistry(Registry.LocalMachine, "SOFTWARE\\MICROSOFT\\PACKAGEMANAGEMENT"))
                 .Concat(GetProvidersFromRegistry(Registry.CurrentUser, "SOFTWARE\\MICROSOFT\\PACKAGEMANAGEMENT"));
 
+#endif
```

Comment:
It's better to use `#if CoreCLR ... #else ... #endif` here, so that in FullCLR you won't call `Enumerable.Empty<string>()` twice. For example:
```
#if CORECLR
      var providerAssemblies = Enumerable.Empty<string>();
#else
      var providerAssemblies = (_initialized ? Enumerable.Empty<string>() : _defaultProviders)
            ....
#endif
```